src/app/components/utility/modal/modal.component.ts

```diff
@@ -137,10 +145,10 @@ export class ModalComponent implements OnInit {
         this.form = this.params['form'];
       }
     }
-    if (this.isEditorRequired) {
+
```

Comment:
Try to make this modal component as generic as possible. You could add `updateSubmissionDetails` as an input. (That is, other components should be dependent on this and not the other way.)

Reply:
okay, then making another variable for only disabling the modal button which will be of type boolean, I think it will be better. Let me know if it looks good to you